The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/5492802108/jobs/10010556889) [(plain)](https://github.com/rust-lang/rust/commit/9f4a7d8c7ca6cba9dd3465483526338bd162ec08/checks/14876000028/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[group]Run git config --global core.autocrlf false
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
  lfs: false
  submodules: false
  set-safe-directory: true
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/should-skip-this.sh
src/ci/scripts/should-skip-this.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-channel.sh
src/ci/scripts/verify-channel.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/collect-cpu-stats.sh
src/ci/scripts/collect-cpu-stats.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-sccache.sh
src/ci/scripts/install-sccache.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/select-xcode.sh
src/ci/scripts/select-xcode.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-clang.sh
src/ci/scripts/install-clang.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-wix.sh
src/ci/scripts/install-wix.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/install-msys2.sh
src/ci/scripts/install-msys2.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-mingw.sh
src/ci/scripts/install-mingw.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/install-ninja.sh
src/ci/scripts/install-ninja.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/enable-docker-ipv6.sh
src/ci/scripts/enable-docker-ipv6.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/disable-git-crlf-conversion.sh
src/ci/scripts/disable-git-crlf-conversion.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/verify-line-endings.sh
src/ci/scripts/verify-line-endings.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
  CACHE_DOMAIN: ci-caches.rust-lang.org
  IMAGE: x86_64-gnu-llvm-14
##[endgroup]
##[group]Run src/ci/scripts/verify-backported-commits.sh
src/ci/scripts/verify-backported-commits.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/verify-stable-version-number.sh
src/ci/scripts/verify-stable-version-number.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
##[group]Run src/ci/scripts/run-build-from-ci.sh
src/ci/scripts/run-build-from-ci.sh
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  PR_CI_JOB: 1
  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
  HEAD_SHA: 9f4a7d8c7ca6cba9dd3465483526338bd162ec08
  SCCACHE_BUCKET: rust-lang-ci-sccache2
  TOOLSTATE_REPO: https://github.com/rust-lang-nursery/rust-toolstate
---
failures:

---- [ui] tests/ui/panics/abort-on-panic.rs#next stdout ----

error in revision `next`: test compilation failed although it shouldn't!
status: exit status: 101
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/panics/abort-on-panic.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--cfg" "next" "-O" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=/checkout/tests/ui=fake-test-src-base" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/panics/abort-on-panic.next/a" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/panics/abort-on-panic.next/auxiliary" "-Ztrait-solver=next"
stdout: none
--- stderr -------------------------------
##[error]error: internal compiler error: compiler/rustc_codegen_llvm/src/context.rs:1010:21: `fn_abi_of_instance(std::sys::unix::stdio::panic_output, [])` failed: Layout(Unknown(impl std::io::Write))
  --> /rustc/FAKE_PREFIX/library/std/src/sys/unix/stdio.rs:95:1
thread 'rustc' panicked at 'Box<dyn Any>', /checkout/compiler/rustc_errors/src/lib.rs:995:33
stack backtrace:
   0:     0x7f2bb155ecf1 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hcc28541cde1e8c96
   1:     0x7f2bb15cd378 - core::fmt::write::h8dd68f136d57a49a
   1:     0x7f2bb15cd378 - core::fmt::write::h8dd68f136d57a49a
   2:     0x7f2bb1553271 - std::io::Write::write_fmt::hc2217b06c7c1b9fa
   3:     0x7f2bb155eb01 - std::sys_common::backtrace::print::h35ddae6db4f80656
   4:     0x7f2bb1561c9a - std::panicking::default_hook::{{closure}}::h8e570f325579d35d
Build completed unsuccessfully in 0:11:00
   5:     0x7f2bb1561989 - std::panicking::default_hook::hb507dc9e16fdd3ce
   6:     0x7f2bb2262f7b - rustc_driver_impl[a691fc3b500bb36b]::install_ice_hook::{closure#0}
   7:     0x7f2bb15623de - std::panicking::rust_panic_with_hook::hc1f372e49006c47e
   8:     0x7f2bb25dc513 - std[ed3d2ba90e1a7828]::panicking::begin_panic::<rustc_errors[20cba4af0f6f6ba9]::ExplicitBug>::{closure#0}
   9:     0x7f2bb25d75a6 - std[ed3d2ba90e1a7828]::sys_common::backtrace::__rust_end_short_backtrace::<std[ed3d2ba90e1a7828]::panicking::begin_panic<rustc_errors[20cba4af0f6f6ba9]::ExplicitBug>::{closure#0}, !>
  10:     0x7f2bb1e4b1f6 - std[ed3d2ba90e1a7828]::panicking::begin_panic::<rustc_errors[20cba4af0f6f6ba9]::ExplicitBug>
  11:     0x7f2bb2667cf5 - <rustc_errors[20cba4af0f6f6ba9]::HandlerInner>::span_bug::<rustc_span[57e6a436add04a83]::span_encoding::Span, alloc[3426741366de250b]::string::String>
  12:     0x7f2bb2667ad9 - <rustc_errors[20cba4af0f6f6ba9]::Handler>::span_bug::<rustc_span[57e6a436add04a83]::span_encoding::Span, alloc[3426741366de250b]::string::String>
  13:     0x7f2bb25dc688 - rustc_middle[6b2bb44635ace8cf]::util::bug::opt_span_bug_fmt::<rustc_span[57e6a436add04a83]::span_encoding::Span>::{closure#0}
  14:     0x7f2bb25dc7fc - rustc_middle[6b2bb44635ace8cf]::ty::context::tls::with_opt::<rustc_middle[6b2bb44635ace8cf]::util::bug::opt_span_bug_fmt<rustc_span[57e6a436add04a83]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  15:     0x7f2bb25db014 - rustc_middle[6b2bb44635ace8cf]::ty::context::tls::with_context_opt::<rustc_middle[6b2bb44635ace8cf]::ty::context::tls::with_opt<rustc_middle[6b2bb44635ace8cf]::util::bug::opt_span_bug_fmt<rustc_span[57e6a436add04a83]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  16:     0x7f2bb1e49647 - rustc_middle[6b2bb44635ace8cf]::util::bug::span_bug_fmt::<rustc_span[57e6a436add04a83]::span_encoding::Span>
  17:     0x7f2bb251ecb1 - <rustc_codegen_llvm[8aeb1642c56507c6]::context::CodegenCx as rustc_middle[6b2bb44635ace8cf]::ty::layout::FnAbiOfHelpers>::handle_fn_abi_err
  18:     0x7f2bb251ea50 - <rustc_codegen_llvm[8aeb1642c56507c6]::builder::Builder as rustc_middle[6b2bb44635ace8cf]::ty::layout::FnAbiOfHelpers>::handle_fn_abi_err
  19:     0x7f2bb250520e - <rustc_codegen_llvm[8aeb1642c56507c6]::builder::Builder as rustc_middle[6b2bb44635ace8cf]::ty::layout::FnAbiOf>::fn_abi_of_instance::{closure#0}
  20:     0x7f2bb251e033 - <rustc_codegen_llvm[8aeb1642c56507c6]::builder::Builder as rustc_middle[6b2bb44635ace8cf]::ty::layout::FnAbiOf>::fn_abi_of_instance
  21:     0x7f2bb2515934 - <rustc_codegen_ssa[9286b1956bbe6f69]::mir::FunctionCx<rustc_codegen_llvm[8aeb1642c56507c6]::builder::Builder>>::codegen_call_terminator
  22:     0x7f2bb250ec2b - <rustc_codegen_ssa[9286b1956bbe6f69]::mir::FunctionCx<rustc_codegen_llvm[8aeb1642c56507c6]::builder::Builder>>::codegen_terminator
  23:     0x7f2bb24f979b - rustc_codegen_ssa[9286b1956bbe6f69]::mir::codegen_mir::<rustc_codegen_llvm[8aeb1642c56507c6]::builder::Builder>
  24:     0x7f2bb24fdb8d - rustc_codegen_ssa[9286b1956bbe6f69]::base::codegen_instance::<rustc_codegen_llvm[8aeb1642c56507c6]::builder::Builder>
  25:     0x7f2bb2606235 - <rustc_middle[6b2bb44635ace8cf]::mir::mono::MonoItem as rustc_codegen_ssa[9286b1956bbe6f69]::mono_item::MonoItemExt>::define::<rustc_codegen_llvm[8aeb1642c56507c6]::builder::Builder>
  26:     0x7f2bb2657f0b - rustc_codegen_llvm[8aeb1642c56507c6]::base::compile_codegen_unit::module_codegen
  27:     0x7f2bb2656cb6 - rustc_codegen_llvm[8aeb1642c56507c6]::base::compile_codegen_unit
  28:     0x7f2bb2520585 - <rustc_codegen_llvm[8aeb1642c56507c6]::LlvmCodegenBackend as rustc_codegen_ssa[9286b1956bbe6f69]::traits::backend::CodegenBackend>::codegen_crate
  29:     0x7f2bb244616f - <rustc_session[25ae73c155ec3365]::session::Session>::time::<alloc[3426741366de250b]::boxed::Box<dyn core[c7cda6c39d66cd81]::any::Any>, rustc_interface[310e2c4c7f90cc3d]::passes::start_codegen::{closure#0}>
  30:     0x7f2bb240e428 - rustc_interface[310e2c4c7f90cc3d]::passes::start_codegen
  31:     0x7f2bb2465cfd - <rustc_middle[6b2bb44635ace8cf]::ty::context::GlobalCtxt>::enter::<<rustc_interface[310e2c4c7f90cc3d]::queries::Queries>::ongoing_codegen::{closure#0}, core[c7cda6c39d66cd81]::result::Result<alloc[3426741366de250b]::boxed::Box<dyn core[c7cda6c39d66cd81]::any::Any>, rustc_span[57e6a436add04a83]::ErrorGuaranteed>>
  32:     0x7f2bb2444cd6 - <rustc_interface[310e2c4c7f90cc3d]::queries::Queries>::ongoing_codegen
  33:     0x7f2bb22d0872 - <rustc_interface[310e2c4c7f90cc3d]::interface::Compiler>::enter::<rustc_driver_impl[a691fc3b500bb36b]::run_compiler::{closure#1}::{closure#2}, core[c7cda6c39d66cd81]::result::Result<core[c7cda6c39d66cd81]::option::Option<rustc_interface[310e2c4c7f90cc3d]::queries::Linker>, rustc_span[57e6a436add04a83]::ErrorGuaranteed>>
  34:     0x7f2bb22b6899 - rustc_span[57e6a436add04a83]::set_source_map::<core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>, rustc_interface[310e2c4c7f90cc3d]::interface::run_compiler<core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>, rustc_driver_impl[a691fc3b500bb36b]::run_compiler::{closure#1}>::{closure#0}::{closure#0}>
  35:     0x7f2bb2287229 - <scoped_tls[1b67669b7709bbb1]::ScopedKey<rustc_span[57e6a436add04a83]::SessionGlobals>>::set::<rustc_interface[310e2c4c7f90cc3d]::interface::run_compiler<core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>, rustc_driver_impl[a691fc3b500bb36b]::run_compiler::{closure#1}>::{closure#0}, core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>>
  36:     0x7f2bb2267bc6 - std[ed3d2ba90e1a7828]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[310e2c4c7f90cc3d]::util::run_in_thread_pool_with_globals<rustc_interface[310e2c4c7f90cc3d]::interface::run_compiler<core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>, rustc_driver_impl[a691fc3b500bb36b]::run_compiler::{closure#1}>::{closure#0}, core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>>
  37:     0x7f2bb22679c8 - std[ed3d2ba90e1a7828]::panicking::try::<core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>, core[c7cda6c39d66cd81]::panic::unwind_safe::AssertUnwindSafe<<std[ed3d2ba90e1a7828]::thread::Builder>::spawn_unchecked_<rustc_interface[310e2c4c7f90cc3d]::util::run_in_thread_pool_with_globals<rustc_interface[310e2c4c7f90cc3d]::interface::run_compiler<core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>, rustc_driver_impl[a691fc3b500bb36b]::run_compiler::{closure#1}>::{closure#0}, core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>>::{closure#1}::{closure#0}>>
  38:     0x7f2bb22bf43f - <<std[ed3d2ba90e1a7828]::thread::Builder>::spawn_unchecked_<rustc_interface[310e2c4c7f90cc3d]::util::run_in_thread_pool_with_globals<rustc_interface[310e2c4c7f90cc3d]::interface::run_compiler<core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>, rustc_driver_impl[a691fc3b500bb36b]::run_compiler::{closure#1}>::{closure#0}, core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[c7cda6c39d66cd81]::result::Result<(), rustc_span[57e6a436add04a83]::ErrorGuaranteed>>::{closure#1} as core[c7cda6c39d66cd81]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  40:     0x7f2bb1305b43 - <unknown>
  41:     0x7f2bb1397a00 - <unknown>
  42:                0x0 - <unknown>


note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.72.0-nightly (3fdd7682c 2023-07-08) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0 -Z trait-solver=next
query stack during panic:
end of query stack
error: aborting due to previous error
------------------------------------------
```

</details>

--- a/compiler/rustc_hir_typeck/src/pat.rs
+++ b/compiler/rustc_hir_typeck/src/pat.rs
@@ -335,8 +335,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         expected: Ty<'tcx>,
         mut def_bm: BindingMode,
     ) -> (Ty<'tcx>, BindingMode) {
-        let mut expected = self.resolve_vars_with_obligations(expected);
-
+        let mut expected = self.try_structurally_resolve_type(pat.span, expected);
         // Peel off as many `&` or `&mut` from the scrutinee type as possible. For example,
         // for `match &&&mut Some(5)` the loop runs three times, aborting when it reaches
         // the `Some(5)` which is not of type Ref.
@@ -353,7 +352,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             // Preserve the reference type. We'll need it later during THIR lowering.
             pat_adjustments.push(expected);
 
-            expected = inner_ty;
+            expected = self.try_structurally_resolve_type(pat.span, inner_ty);
             def_bm = ty::BindByReference(match def_bm {
                 // If default binding mode is by value, make it `ref` or `ref mut`
                 // (depending on whether we observe `&` or `&mut`).
@@ -627,6 +626,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         local_ty
     }
 
+    /// When a variable is bound several times in a `PatKind::Or`, it'll resolve all of the
+    /// subsequent bindings of the same name to the first usage. Verify that all of these
+    /// bindings have the same type by comparing them all against the type of that first pat.
     fn check_binding_alt_eq_ty(
         &self,
         ba: hir::BindingAnnotation,
@@ -638,7 +640,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let var_ty = self.local_ty(span, var_id);
         if let Some(mut err) = self.demand_eqtype_pat_diag(span, var_ty, ty, ti) {
             let hir = self.tcx.hir();
-            let var_ty = self.resolve_vars_with_obligations(var_ty);
+            let var_ty = self.resolve_vars_if_possible(var_ty);
             let msg = format!("first introduced with type `{var_ty}` here");
             err.span_label(hir.span(var_id), msg);
             let in_match = hir.parent_iter(var_id).any(|(_, n)| {
@@ -656,7 +658,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 &mut err,
                 span,
                 var_ty,
-                self.resolve_vars_with_obligations(ty),
+                self.resolve_vars_if_possible(ty),
                 ba,
             );
             err.emit();

--- a/compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs
+++ b/compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs
@@ -1640,6 +1640,8 @@ impl<'tcx> InferCtxtPrivExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
                     )
                     .into(),
                 };
+                // FIXME(-Ztrait-solver=next): For diagnostic purposes, it would be nice
+                // to deeply normalize this type.
                 let normalized_term =
                     ocx.normalize(&obligation.cause, obligation.param_env, unnormalized_term);
 

--- a/tests/ui/panics/abort-on-panic.rs
+++ b/tests/ui/panics/abort-on-panic.rs
@@ -1,4 +1,6 @@
 // run-pass
+// revisions: current next
+//[next] compile-flags: -Ztrait-solver=next
 
 #![allow(unused_must_use)]
 #![feature(c_unwind)]

